src/prefect/engine/cloud/task_runner.py

```diff
@@ -111,10 +112,22 @@ def call_runner_target_handlers(self, old_state: State, new_state: State) -> Sta
             cloud_state = new_state
             state = self.client.set_task_run_state(
                 task_run_id=task_run_id,
-                version=version,
+                version=version if cloud_state.is_running() else None,
                 state=cloud_state,
                 cache_for=self.task.cache_for,
             )
+        except VersionLockError as exc:
+            state = self.client.get_task_run_state(task_run_id=task_run_id)
+
+            if state.is_running():
+                raise ENDRUN(state=state)
+
+            self.logger.debug(
+                "Version lock encountered for task {}, proceeding with state {}...".format(
+                    self.task.name, type(state).__name__
+                )
+            )
+            new_state = state
```

Comment:
we probably should populate the result here as well (it's generally assumed that results are populated at the beginning of a run and this could occur anywhere).  I think something like:
```python
new_state = state.load_result(self.task.result)
```
should work (you'll want to confirm my syntax).

Not overcomplicate, but this might need to live inside it's own `try / except` as well, and if an error occurs we probably need to raise `ENDRUN`

Reply:
Yeah good call 👍

Reply:
Instead of `self.task.result` I'm going to do the `self.result` since at initialization of the task runner it is set up to use the Flow's result if none was set on the task:
https://github.com/PrefectHQ/prefect/blob/71d40d7a4d3f97d0ca562b80b9033646e1a2c9ae/src/prefect/engine/task_runner.py#L83-L86